cleanup/cleanup.sh

```diff
@@ -1,3 +1,111 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
-echo "Plugin called."
+IMAGE_NAME=$1
+
+set -e -o pipefail -u
+
+GIT_COMMIT_LIMIT=${KUBECTL_PLUGINS_LOCAL_FLAG_GIT_COMMIT_LIMIT}
+GIT_REPO_PATH=${KUBECTL_PLUGINS_LOCAL_FLAG_GIT_REPO_PATH}
+FORCE=${KUBECTL_PLUGINS_LOCAL_FLAG_FORCE}
+KEEP=${KUBECTL_PLUGINS_LOCAL_FLAG_KEEP}
+NAMESPACE=${KUBECTL_PLUGINS_CURRENT_NAMESPACE}
+oc="${KUBECTL_PLUGINS_CALLER} -n ${NAMESPACE}"
+
+get_commit_list() {
+    # all commits except HEAD in the Git repository
+    git -C ${GIT_REPO_PATH} rev-list --max-count ${GIT_COMMIT_LIMIT} --author-date-order HEAD~
+}
+
+get_imagestreamtags() {
+    local imagestream_path='{range .status.tags[*]}{.tag}{"\n"}{end}'
+
+    ${oc} get imagestream ${IMAGE_NAME} -o jsonpath="${imagestream_path}" | sort --unique
+}
+
+get_active_imagestreamtags() {
+    local dc_rc_sfs_path='{range .items[*]}{.spec.template.spec.containers[*].image}{"\n"}{end}'
+    local pod_cont_path='{range .items[*]}{.spec.containers[*].image}{"\n"}{end}'
+    local pod_init_path='{range .items[*]}{.spec.initContainers[*].image}{"\n"}{end}'
+    local cron_path='{range .items[*]}{.spec.jobTemplate.spec.template.spec.containers[*].image}{"\n"}{end}'
+
+    local images_in_dc_rc_sset=$(${oc} get dc,rc,statefulset -o jsonpath="${dc_rc_sfs_path}")
+    local images_in_pod_cont=$(${oc} get pod -o jsonpath="${pod_cont_path}")
+    local images_in_pod_init=$(${oc} get pod -o jsonpath="${pod_init_path}")
+    local images_in_cron=$(${oc} get cronjob -o jsonpath="${cron_path}")
+
+    echo "${images_in_dc_rc_sset}" \
+         "${images_in_pod_cont}" \
+         "${images_in_pod_init}" \
+         "${images_in_cron}" \
+         | xargs | tr ' ' '\n' \
+         | grep "/${IMAGE_NAME}:" \
+         | sed "s#.*/${NAMESPACE}/${IMAGE_NAME}:##" \
+         | sort --unique
+}
+
+get_inactive_imagestreamtags() {
+    local all_tags=$(get_imagestreamtags)
+    local all_active=$(get_active_imagestreamtags)
+
+    # all_inactive := all_tags - all_active
+    comm -23 \
+        <(echo ${all_tags} | tr ' ' '\n') \
+        <(echo ${all_active} | tr ' ' '\n')
+}
+
+get_deletion_candidates() {
+    local commit_list=($(get_commit_list))
```

Comment:
Unquoted command expansion may cause unwanted splitting and glob expansion. Use

    mapfile -t commit_list < <(get_commit_list)

Here and later in the code

Reply:
Actually, some set operations could be used here, ie

    comm -12 <(get_commit_list) <(get_inactive_imagestreamtags)

which will output elemets common in both outputs

Reply:
I'm not sure I'm following you. The code above is creating a list of elements. Why would we have to do set  operations?